api/v1beta2/awsmachine_types.go

```diff
@@ -188,7 +200,11 @@ type AWSMachineSpec struct {
 	// Tenancy indicates if instance should run on shared or single-tenant hardware.
 	// +optional
 	// +kubebuilder:validation:Enum:=default;dedicated;host
-	Tenancy string `json:"tenancy,omitempty"`
+	Tenancy Tenancy `json:"tenancy,omitempty"`
+
+	// DedicatedHostPlacement denotes the placement settings for the instance when tenancy=host
+	// +optional
+	DedicatedHostPlacement *DedicatedHostPlacement `json:"dedicatedHostPlacment,omitempty"`
```

Comment:
I'd call this hostPlacement given the enum has both terms separately "Dedicated" and "Host" DedicatedHostPlacement seems confusing to me.

Reply:
renamed.